src/bentoml_cli/bentos.py

```diff
@@ -278,22 +285,72 @@ def push(bento_tag: str, force: bool, threads: int, context: str) -> None:  # ty
     @cli.command()
     @click.argument("build_ctx", type=click.Path(), default=".")
     @click.option(
-        "-f", "--bentofile", type=click.STRING, default=DEFAULT_BENTO_BUILD_FILE
+        "-f",
+        "--bentofile",
+        type=click.STRING,
+        default=DEFAULT_BENTO_BUILD_FILE,
+        help="Path to bentofile. Default to 'bentofile.yaml'",
+    )
+    @click.option(
+        "--version",
+        type=click.STRING,
+        default=None,
+        help="Bento version. By default the version will be generated.",
+    )
+    @click.option(
+        "-o",
+        "--output",
+        type=click.Choice(["machine", "default"]),
+        default="default",
+        show_default=True,
+        help="Output log format. '-o machine' to display only bento tag.",
     )
-    @click.option("--version", type=click.STRING, default=None)
-    def build(build_ctx: str, bentofile: str, version: str) -> None:  # type: ignore (not accessed)
+    @inject
+    def build(  # type: ignore (not accessed)
+        build_ctx: str,
+        bentofile: str,
+        version: str,
+        output: t.Literal["machine", "default"],
+        _bento_store: BentoStore = Provide[BentoMLContainer.bento_store],
+    ) -> None:
         """Build a new Bento from current directory."""
         if sys.path[0] != build_ctx:
             sys.path.insert(0, build_ctx)
 
-        bento = build_bentofile(bentofile, build_ctx=build_ctx, version=version)
+        try:
+            bentofile = resolve_user_filepath(bentofile, build_ctx)
+        except FileNotFoundError:
+            raise bentoml.exceptions.InvalidArgument(
+                f'bentofile "{bentofile}" not found'
+            )
+
+        with open(bentofile, "r", encoding="utf-8") as f:
+            build_config = BentoBuildConfig.from_yaml(f)
+
+        bento = Bento.create(
+            build_config=build_config,
+            version=version,
+            build_ctx=build_ctx,
+        ).save(_bento_store)
+
+        if output == "machine":
+            click.echo(bento.tag)
+            return
+
         click.echo(BENTOML_FIGLET)
-        click.secho(f"Successfully built {bento}.", fg="green")
+
+        # NOTE: Don't change the following message, since it is used by 'bentoml build' to parse for the bento.tag
+        click.secho(f"Successfully built {bento.tag}", fg="green")
+
         click.secho(
             f"\nPossible next steps:\n\n * Containerize your Bento with `bentoml containerize`:\n    $ bentoml containerize {bento.tag}",
-            fg="yellow",
+            fg="blue",
         )
         click.secho(
             f"\n * Push to BentoCloud with `bentoml push`:\n    $ bentoml push {bento.tag}",
-            fg="yellow",
+            fg="blue",
         )
+
+        # We need to pop the build_ctx from sys.path to avoid conflicts.
+        if sys.path[0] == build_ctx:
```

Comment:
Hm, why? Isn't the point of doing this in a subprocess that we shouldn't care?

Also, this should probably be done in the build function itself and not here.

Reply:
This is now the build function right? since from our python API we will just call this.

Reply:
Also do you remember why we do L317?

Reply:
Sorry, to be clear, I think this cleanup should be happening where we modify the syspath and not as an extra step in the CLI.

Reply:
This cleanup should be here right, since we modify the sys.path from the beginning, and then after the _save we should cleanup the sys.path

Reply:
Oh, I somehow forgot that that was being done. That seems odd, why's it here again?

Reply:
If users do the following in service

```python
from src.path import module

```

We need to put the build context into sys.path[0] to be able to find src.path